packages/features/auth/lib/next-auth-options.ts

```diff
@@ -735,16 +766,25 @@ export const AUTH_OPTIONS: AuthOptions = {
           return "/auth/error?error=use-identity-login";
         }
 
+        // Associate with organization if enabled by flag
+        const [orgUsername, orgId] = await checkIfUserShouldBelongToOrg(user.email);
+
         const newUser = await prisma.user.create({
           data: {
             // Slugify the incoming name and append a few random characters to
             // prevent conflicts for users with the same name.
-            username: usernameSlug(user.name),
+            username: orgId ? orgUsername : usernameSlug(user.name),
             emailVerified: new Date(Date.now()),
             name: user.name,
             email: user.email,
             identityProvider: idP,
             identityProviderId: account.providerAccountId,
+            ...(orgId && {
+              organization: { connect: { id: orgId } },
+              teams: {
+                create: { role: MembershipRole.MEMBER, accepted: true, team: { connect: { id: orgId } } },
+              },
+            }),
```

Comment:
We need to link the org by id to the newly created user as well as creating a membership for the user, only if orgId exists which in turn means autolink is turned on